DatadogSessionReplay/Sources/Recorder/Utilities/UIImage+SessionReplay.swift

```diff
@@ -37,7 +37,12 @@ extension DatadogExtension where ExtendedType: UIImage {
 
 extension UIColor {
     var srIdentifier: String {
-        return "\(hash)"
+        var r: CGFloat = 0
+        var g: CGFloat = 0
+        var b: CGFloat = 0
+        var a: CGFloat = 0
+        getRed(&r, green: &g, blue: &b, alpha: &a)
+        return String(format: "%02X%02X%02X%02X", Int(round(r * 255)), Int(round(g * 255)), Int(round(b * 255)), Int(round(a * 255)))
```

Comment:
question/ What was the actual issue with using `hash`? Can we elaborate on this in PR description?

What is unclear to me:

> Fixes the SR color identifier, which is causing unnecessary upload of resources
> ...
> different versions of iOS produce different images, and hashes

So, if the `hash` is stable on iOS version, why  was it causing non-deterministic resource IDs?

Reply:
It appears to be stable only in the developer environment. It's totally random on production.

Reply:
Ok, pretty clear 👌